app/helpers/catalog_helper.rb

```diff
@@ -145,42 +145,45 @@ def document_icon( document )
     resultIcon = 'glyphicon-file'
 
     if document['files_tesim'] != nil
-
       # SIMPLE OBJECT: look for preview image or format-appropriate icon
-      fieldData = document['files_tesim']
-      simpleUse = nil
-
-      if fieldData != nil
-        fieldData.each do |datum|
-          files = JSON.parse(datum)
-          if files["use"].end_with?("-service")
-            simpleUse = files["use"]
-            break
-          end
-        end
-      end
+      field_data = document['files_tesim']
+      simpleUse = file_use_type (field_data)

```

Comment:
I am not sure whether it's sufficient to determine a simple object with the files_tesim field and a -service use of a existing file or not. There is a use case that DOMM user would like to add files to the top level of a complex object. A similar case they've done so far is just ingest a file in the first component, and along with it are other child components, see http://library.ucsd.edu/dc/object/bb1162536b (curator access) . Otherwise 👍 .